Makefile

```diff
@@ -0,0 +1,29 @@
+
+NAME	= go-recipes
+BIN 	= go-recipes
+
+all: build
+
+test: deps
+	go test -timeout=5s -cover -race -v ./...
+
+ci_test: test
+
+build: deps test
+	go build -v -a -o $(NAME) .
+
+run:
+	./$(BIN)
+
+deps:
+	go get github.com/stretchr/testify/require
+	go get ./...
+
```

Comment:
You can use -t flag to get test dependencies: go get -t ./...
"The -t flag instructs get to also download the packages required to build
the tests for the specified packages."